packages/grid/src/vaadin-grid.js

```diff
@@ -690,8 +689,20 @@ class Grid extends ElementMixin(
     const slot = document.createElement('slot');
     slot.setAttribute('name', slotName);
 
-    cell.appendChild(slot);
+    if (column && column._focusButtonMode) {
+      slot.setAttribute('role', 'button');
+      slot.setAttribute('tabindex', '-1');
+
+      // Patch `focus()` to use the slot
+      cell._focusButton = slot;
```

Comment:
Since we are not using a button anymore `_focusButton` can be confusing. How about `_focusElement`?

Reply:
Reverted the commit to use the original approach, as we decided that using `<slot>` is not a proper solution (some a11y checking tools might complain about that).